Jenkinsfile

```diff
@@ -297,3 +350,14 @@ def installAzureCli() {
 	sh "sudo apt update"
 	sh "sudo apt install -y azure-cli"
 }
+
+def boolean docsFileOnly() {
+    if (env.CHANGE_TARGET == null) {
+        return false;
+    }
+
+    return sh(
+        returnStatus: true,
+        script: "git diff --name-only origin/${env.CHANGE_TARGET} | grep -i -v '\\.md'"
```

Comment:
Why the case independence?

Reply:
I don't think this works if the PR is not rebased on the branch and if .e.g. main has other changes because this will show the diff between your branch and main. You want to show the diff between you branch and the most recent common parent:

https://issues.jenkins.io/browse/JENKINS-54285?focusedCommentId=354715&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-354715

Reply:
> Why the case independence?

Well I thought it'd bring a bit more flexibility, but if you don't think that's bringing anything, I'll remove it.

Reply:
> I don't think this works if the PR is not rebased on the branch and if .e.g. main has other changes because this will show the diff between your branch and main. You want to show the diff between you branch and the most recent common parent:
> 
> https://issues.jenkins.io/browse/JENKINS-54285?focusedCommentId=354715&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-354715

Ah yes that's a good point. So IIUC using `git rev-parse origin/main` for instance will give me the common ancestor between `HEAD` and `origin/main`, right?